datanode/api/trading_data_v2.go

```diff
@@ -3048,6 +3055,102 @@ func (t *tradingDataServiceV2) GetVegaTime(ctx context.Context, _ *v2.GetVegaTim
 
 // -- NetworkHistory --.
 
+func (t *tradingDataServiceV2) ExportNetworkHistory(req *v2.ExportNetworkHistoryRequest, stream v2.TradingDataService_ExportNetworkHistoryServer) error {
+	defer metrics.StartAPIRequestAndTimeGRPC("ExportNetworkHistory")()
+
+	if req.Table == v2.Table_TABLE_UNSPECIFIED {
+		return formatE(ErrNetworkHistoryNoTableName, errors.New("empty table name"))
+	}
+
+	tableName := strings.TrimPrefix(strings.ToLower(req.Table.String()), "table_")
+
+	segments, err := t.networkHistoryService.ListAllHistorySegments()
+	if err != nil {
+		return formatE(ErrListAllNetworkHistorySegment, err)
+	}
+
+	ch, err := networkhistory.GetContiguousHistory(segments, req.FromBlock, req.ToBlock)
+	if err != nil {
+		return formatE(ErrNetworkHistoryGetContiguousSegments, err)
+	}
+
+	header := metadata.Pairs("Content-Disposition", fmt.Sprintf("attachment;filename=vega-%s-%06d-%06d.csv", tableName, ch.HeightFrom, ch.HeightTo))
+	if err := stream.SendHeader(header); err != nil {
+		return formatE(ErrSendingGRPCHeader, err)
+	}
+
+	var chunk bytes.Buffer
+	var csvHeaders string
+	rowNum := 1
+
+	for _, segment := range ch.SegmentsOldestFirst {
+		segmentReader, err := t.networkHistoryService.GetHistorySegmentReader(stream.Context(), segment.GetHistorySegmentId())
+		if err != nil {
+			segmentReader.Close()
+			return formatE(ErrNetworkHistoryOpeningSegment, err)
+		}
+
+		segmentData, err := fsutil.ReadNetworkHistorySegmentData(segmentReader, tableName)
+		if err != nil {
+			segmentReader.Close()
+			return formatE(ErrNetworkHistoryExtractingSegment, err)
+		}
+
+		scanner := bufio.NewScanner(segmentData)
+		firstRowInSegment := true
+		for ; scanner.Scan(); rowNum++ {
+			row := scanner.Text()
+			if rowNum == 1 {
+				csvHeaders = row
+			}
+
+			// Don't repeat headers if they're the same in subsequent segments
+			if firstRowInSegment && rowNum != 1 && csvHeaders == row {
+				firstRowInSegment = false
+				continue
+			}
+
+			// If we're in a subsequent segment and headers have changed, write a blank line before the new headers
+			if firstRowInSegment && rowNum != 1 && csvHeaders != row {
+				chunk.WriteString("\n")
+				csvHeaders = row
+			}
+
+			firstRowInSegment = false
+			row = strings.Replace(row, `\x`, "", -1) // get rid of the \x prefix from BYTEA columns
```

Comment:
Do we want to do this?  wouldn't be better to just leave it in the format postgres understands for now (which may be beneficial) until we learn otherwise?  any end user can handle this easily enough

Reply:
Yeah I dunno, I was kind of torn - took it out, then put it back again.. I was just imagining having to repeatedly answer 'what is this \x for' - but this solution is kind of crappy, e.g. it would remove \x from any text data as well. 

I'll remove it and if we get complains we can always revisit. To do it right it would know the column types and only do it for bytea columns but then we have to get into parsing the CSV data and making sure we do all the escaping right etc.. which I was tyring to avoid.